### ISSUE TITLE
Returning from main()

### ISSUE THREAD

**Issue by Scotsgeek** [RunestoneInteractive/rs#496]
**Please check the current issues** Many bug reports are duplicates, this just creates more work for us. Searching the issues first may give you your answer or a workaround! If not adding new information to an existing report is much more helpful than a new report.

**What Course are you in**
Thinkcpp

**What Page were you on**
5.9 Returning from main

**What is your username**
None

**Describe the bug**
"The usual return value from main is 0, which indicates that the program succeeded at whatever it was supposed to do. If something goes wrong, **it is common to return -1**, or some other value that indicates what kind of error occurred."

Not correct!

In stdlib.h, two macros are defined that are commonly used to return from main() or any other function:

"#define EXIT_SUCCESS 0"
"#define EXIT_FAILURE 1"

-1 is used for other purposes such as EOF, or "End Of File", or some other error condition.

See: [man EOF](https://man7.org/linux/man-pages/man3/EOF.3const.html)

**Traceback**
None

**Javascript Errors**
None

**Comment by bnmnetp** [comment]
You're right that the wording is misleading — the exit status is truncated to 8 bits, so `return -1` reaches the shell as 255, and `EXIT_SUCCESS`/`EXIT_FAILURE` are the portable spellings.

This is book prose, though, and thinkcpp's source lives in its own repo: https://github.com/RunestoneInteractive/thinkcpp. Closing here; a PR against that repo (or an issue there) is the way to get it changed, and it's a small edit if you're willing.
